sound/soc/sof/sof-acpi-dev.c

```diff
@@ -127,13 +161,26 @@ static const struct dev_pm_ops sof_acpi_pm = {
 
 static void sof_acpi_probe_complete(struct device *dev)
 {
+	int i;
+
 	if (sof_acpi_debug & SOF_ACPI_DISABLE_PM_RUNTIME)
 		return;
 
 	/* allow runtime_pm */
 	pm_runtime_set_autosuspend_delay(dev, SND_SOF_SUSPEND_DELAY_MS);
 	pm_runtime_use_autosuspend(dev);
+	pm_runtime_set_active(dev);
 	pm_runtime_enable(dev);
+	pm_runtime_mark_last_busy(dev);
+
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_BAYTRAIL)
+	/* set driver data in the quirk table */
+	for (i = 0; i < ARRAY_SIZE(minnowbard_quirk_table); i++)
+		minnowbard_quirk_table[i].driver_data = dev;
```

Comment:
this is a bit ugly, no? You could just check the quirk, which sets a variable and then you test said variable.

Reply:
and you can also use the sof-acpi-debug kernel parameter to override the DMI and also disable PM!